kernel/sockdev.c

```diff
@@ -309,9 +309,12 @@ static int _sd_sendmsg(
     struct msghdr msg_buf;
     const struct msghdr* msg_ptr;
 
-    if (!sd || !_valid_sock(sock) || !msg)
```

Comment:
The correct error number should be

>**EFAULT**
An invalid user space address was specified for an argument. 

instead of

>EINVAL
Invalid argument passed.